File: src-1.x/BYTES.NET/WPF/Converters/BoolInvertConverter.vb
Imports System.Globalization

Namespace WPF.Converters

    ''' <summary>
    ''' converter inverting a boolean value
    ''' </summary>
    Public Class BoolInvertConverter

        Implements IValueConverter

        Public Function Convert(value As Object, targetType As Type, parameter As Object, culture As CultureInfo) As Object Implements IValueConverter.Convert

            Try

                If CBool(value) Then

                    Return False

                End If

                Return True

            Catch ex As Exception

                Return False

            End Try

        End Function

        Public Function ConvertBack(value As Object, targetType As Type, parameter As Object, culture As CultureInfo) As Object Implements IValueConverter.ConvertBack

            Try

                If CBool(value) Then

                    Return False

                End If

                Return True

            Catch ex As Exception

                Return False

            End Try

        End Function

    End Class

End Namespace

File: src-2023/BYTES.NET.WPF/Converters/BoolInvertConverter.cs
//import .net namespace(s) required
using System;
using System.Collections.Generic;
using System.Globalization;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Windows.Data;

namespace BYTES.NET.WPF.Converters
{
    public class BoolInvertConverter : IValueConverter
    {
        public object Convert(object value, Type targetType, object parameter, CultureInfo culture)
        {
            try
            {
                if (value is bool && (bool)value)
                {
                    return false;
                }

                return true;
            }
            catch (Exception ex)
            {
                return false;
            }
        }

        public object ConvertBack(object value, Type targetType, object parameter, CultureInfo culture)
        {
            try
            {
                if (value is bool && (bool)value)
                {
                    return false;
                }

                return true;
            }
            catch (Exception ex)
            {
                return false;
            }
        }
    }
}